grails-app/domain/com/redhat/theses/Link.groovy

```diff
@@ -0,0 +1,15 @@
+package com.redhat.theses
+
+class Link implements Serializable{
+    String url
+    
+    static mapping = {
+        id composite: ['url']
```

Comment:
This is not a good idea. If you try to add the same link into another thesis, it will probably fail, unless you check for an existing link. Even if you check in the controller for an existing link, it's still a bad idea. In the future, we might want to extend the Link class to include description or something like, that would make it impossible to add two same links to two different thesis with different description. So my suggestion is to make the id not a composite of an URL, but just an incremental number.